File: src/main/resources/output/IFC4X3_js/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcElementComponentType/IfcReinforcingElementType/IfcTendonConduitType/IfcTendonConduitType.js
class IfcTendonConduitType extends IfcReinforcingElementType {
    constructor() {
        /** @type {IFCTENDONCONDUITTYPEENUM} */
        this.PredefinedType = null;
    }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcTendonConduitTypeEnum.USERDEFINED) OR ((PredefinedType = IfcTendonConduitTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}

File: src/main/resources/output/IFC4X3_python/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcElementComponentType/IfcReinforcingElementType/IfcTendonConduitType/IfcTendonConduitType.py
class IfcTendonConduitType(IfcReinforcingElementType):
    def __init__(self):
        self.PredefinedType: IfcTendonConduitTypeEnum = None

    # === WHERE CLAUSES ===
    # CorrectPredefinedType : (PredefinedType <> IfcTendonConduitTypeEnum.USERDEFINED) OR ((PredefinedType = IfcTendonConduitTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
